test/e2e/apimachinery/flowcontrol.go

```diff
@@ -247,6 +252,224 @@ var _ = SIGDescribe("API priority and fairness", func() {
+
+		ginkgo.By("updating")
+		fsToUpdate := fsPatched.DeepCopy()
+		fsToUpdate.Annotations["updated"] = "true"
+		fsUpdated, err := client.Update(ctx, fsToUpdate, metav1.UpdateOptions{})
+		framework.ExpectNoError(err)
+		gomega.Expect(fsUpdated.Annotations).To(gomega.HaveKeyWithValue("updated", "true"), "updated object should have the applied annotation")
+
+		framework.Logf("waiting for watch events with expected annotations")
+		for sawAnnotation := false; !sawAnnotation; {
+			select {
+			case evt, ok := <-fsWatch.ResultChan():
+				if !ok {
+					framework.Fail("watch channel should not close")
+				}
+				gomega.Expect(evt.Type).To(gomega.Equal(watch.Modified))
+				fsWatched, isFS := evt.Object.(*flowcontrol.FlowSchema)
+				if !isFS {
+					framework.Failf("expected an object of type: %T, but got %T", &flowcontrol.FlowSchema{}, evt.Object)
+				}
+				if fsWatched.Annotations["patched"] == "true" {
+					sawAnnotation = true
+					fsWatch.Stop()
+				} else {
+					framework.Logf("missing expected annotations, waiting: %#v", fsWatched.Annotations)
+				}
+			case <-time.After(wait.ForeverTestTimeout):
+				framework.Fail("timed out waiting for watch event")
+			}
+		}
+
+		ginkgo.By("getting /status")
+		resource := flowcontrol.SchemeGroupVersion.WithResource("flowschemas")
+		fsStatusRead, err := f.DynamicClient.Resource(resource).Get(ctx, fsCreated.Name, metav1.GetOptions{}, "status")
+		framework.ExpectNoError(err)
+		gomega.Expect(fsStatusRead.GetObjectKind().GroupVersionKind()).To(gomega.Equal(flowcontrol.SchemeGroupVersion.WithKind("FlowSchema")))
+		gomega.Expect(fsStatusRead.GetUID()).To(gomega.Equal(fsCreated.UID))
+
+		ginkgo.By("patching /status")
+		fsStatusPatched, err := client.Patch(ctx, fsCreated.Name, types.MergePatchType,
+			[]byte(`{"metadata":{"annotations":{"patchedstatus":"true"}},"status":{"conditions":[{"type":"PatchStatusFailed","status":"False","reason":"e2e"}]}}`), metav1.PatchOptions{}, "status")
+		framework.ExpectNoError(err)
+		condition := apihelpers.GetFlowSchemaConditionByType(fsStatusPatched, flowcontrol.FlowSchemaConditionType("PatchStatusFailed"))
+		gomega.Expect(condition).NotTo(gomega.BeNil())
+		gomega.Expect(fsStatusPatched.Annotations).To(gomega.HaveKeyWithValue("patchedstatus", "true"), "patched object should have the applied annotation")
+
+		ginkgo.By("updating /status")
+		fsStatusToUpdate := fsStatusPatched.DeepCopy()
+		fsStatusToUpdate.Status.Conditions = append(fsStatusToUpdate.Status.Conditions, flowcontrol.FlowSchemaCondition{
+			Type:    "StatusUpdateFailed",
+			Status:  flowcontrol.ConditionFalse,
+			Reason:  "E2E",
+			Message: "Set from an e2e test",
+		})
+		fsStatusUpdated, err := client.UpdateStatus(ctx, fsStatusToUpdate, metav1.UpdateOptions{})
+		framework.ExpectNoError(err)
+		condition = apihelpers.GetFlowSchemaConditionByType(fsStatusUpdated, flowcontrol.FlowSchemaConditionType("StatusUpdateFailed"))
+		gomega.Expect(condition).NotTo(gomega.BeNil())
```

Comment:
same question about whether another client can update this and cause a conflict we have to retry on

Reply:
taken care of